### PR TITLE
Validate path for inappropriate node types.

### PR DESCRIPTION
**Pull request by rmclaren**
## Description

We were not checking to make sure the query path string does not contain node types it should not (like non-repeating sequences). The paths would be recognized but lead to a crash instead of warning the user about the invalid path for the subset. This pull request aims to fix this...

## Issues

#1427

### CODE REVIEW

**Review by srherbener — APPROVED** [review]
Thanks for fixing this! Tested on my Mac and all ioda-converter tests passed.

**Comment by PatNichols** [comment]
@rmclaren Could you explain the problem more in depth.

File: test/testinput/bufr_ncep_prepbufr_adpsfc.yaml [hunk 1]
```diff
@@ -24,6 +24,8 @@ observations:
             query: "*/XOB"
           latitude:
             query: "*/YOB"
+          invalidLongitude:  # make sure that path strings can't contain sequences
```
Comment (BenjaminRuston) [inline test/testinput/bufr_ncep_prepbufr_adpsfc.yaml:27]: @rmclaren is this something that does not exist and tests that it handles it gracefully?

Comment (rmclaren) [reply test/testinput/bufr_ncep_prepbufr_adpsfc.yaml:27]: It points to something that exists but the query string includes the non-repeating sequence mnemonic which should not be part of the query string... The old version of the code will crash if you do this (the fixed version detects the issue and warns the user without crashing).

Comment (BenjaminRuston) [reply test/testinput/bufr_ncep_prepbufr_adpsfc.yaml:27]: oh, that is subtle but now follow though `XOB` is in the `HEADR` that latter sequence mnemonic should not be in the query

yes not sure how to express this any better in a succinct line

**Review by BenjaminRuston — APPROVED** [review]
i follow what this is doing now thanks @rmclaren for the explanation and for the catch

**Pull request merged**